nilearn/signal.py

```diff
@@ -829,6 +830,8 @@ def _handle_scrubbed_volumes(
     """Interpolate or censor scrubbed volumes."""
     if sample_mask is None:
         return signals, confounds, sample_mask
+    elif sample_mask.size == 0:
```

Comment:
My understanding is sample_mask can be an array with 0's, in which case its size is nonzero even if all volumes are removed ?

Reply:
[Here](https://github.com/nilearn/nilearn/blob/main/nilearn/interfaces/fmriprep/load_confounds_scrub.py#L129) it looks like sample_mask contains the indices of the volumes which are *not* outliers. So the size is only nonzero if at least one volume is not an outlier. Does it make sense?

Reply:
@victoris93 is correct here with how this is `sample_mask` implemented: it's either `None`, meaning no scrubbing, or index of volumes kept.
However this does seem to confuse people quite a lot....

Reply:
OK, I understood that the parameter has been coerced to this form when it reaches this function. It is hard to get this from the code.